### ISSUE TITLE
Uncaught (in promise) Error: InvalidStateError

### ISSUE THREAD

**Issue by rajpurohityogesh** [coding-with-chaim/group-video-final#14]
Hello guys I am facing the following issue when I try to join a meeting that is already started.

At meeting Joiner Side:
Uncaught (in promise) Error: InvalidStateError: Failed to execute 'setRemoteDescription' on 'RTCPeerConnection': Failed to set remote answer sdp: Called in wrong state: stable
    at makeError (index.js:17)
    at index.js:216

At meeting create side:
Uncaught Error: [object RTCErrorEvent]
    at makeError (index.js:17)
    at RTCDataChannel._channel.onerror (index.js:490)

It's been 4 days and I probably tried every possible way. **Everyone** and **@coding-with-chaim** it will be so helpful if you can have a look at the error and possibly tell me what I am doing wrong.

**Comment by ChaimF90** [comment]
which version of simple peer are you using? you can check by running the command `npm ls simple-peer`

**Comment by rajpurohityogesh** [comment]
I first tried the latest version but then also I was facing the same issue but later I changed It to 9.6.2 as it was used in the @coding-with-chaim 's code so.

**Comment by ChaimF90** [comment]
are you having this issue still with 9.6.2?

**Comment by rajpurohityogesh** [comment]
Yes, right now I am trying it again with the latest version 9.11.0

**Comment by rajpurohityogesh** [comment]
Did you faced the same issue too ?

**Comment by ChaimF90** [comment]
I am coding-with-chaim just using a diff account. Yes I have had this issue I believe and for me using 9.6.2 was the answer. Are you using this code exactly, or are you writing your code following this code as a guide?

**Comment by rajpurohityogesh** [comment]
Yes sir I am using this code as a guide. I from the home page you can create a meet or join a meet. And in the meeting page code, I am doing all the peer creation and signal assigning.

**Comment by ChaimF90** [comment]
ok so if I had to guess there is a bug in your code and the issue is not with simple peer.

**Comment by rajpurohityogesh** [comment]
Ok sir so can you tell me where it might be because you know the core wrtc so I gess you might know the error terms.

**Comment by rajpurohityogesh** [comment]
This is what I am facing sir. I am getting a black box for other users' videos and getting the following error in the console.

![image](https://user-images.githubusercontent.com/67944765/119559478-5a393080-bdc0-11eb-8075-25c319caf28d.png)

**Comment by rajpurohityogesh** [comment]
Sir, I guess I have caught the main issue but don't know how to resolve it.

When we get a signal back from another user on the "returning signal" event and we emit the "receiving returned signal" event it is getting fired twice.

Do you have any idea why this is happening!!!!!!!
